File: archived/ContextMenu/js/js/scenario1.js
//// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
//// ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
//// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//// PARTICULAR PURPOSE.
////
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";
    var page = WinJS.UI.Pages.define("/html/scenario1.html", {
        ready: function (element, options) {
            document.getElementById("attachment").addEventListener("contextmenu", attachmentHandler, false);
        }
    });

    // Converts from page to WinRT coordinates, which take scale factor into consideration.
    function pageToWinRT(pageX, pageY) {
        var zoomFactor = document.documentElement.msContentZoomFactor;
        return { x: (pageX - window.pageXOffset) * zoomFactor,
                 y: (pageY - window.pageYOffset) * zoomFactor };
    }

    function attachmentHandler(e) {
        // Create a menu and add commands with callbacks.
        var menu = new Windows.UI.Popups.PopupMenu();
        menu.commands.append(new Windows.UI.Popups.UICommand("Open with", onOpenWith));
        menu.commands.append(new Windows.UI.Popups.UICommand("Save attachment", onSaveAttachment));

        // We don't want to obscure content, so pass in the position representing the selection area.
        // We registered command callbacks; no need to handle the menu completion event
        menu.showAsync(pageToWinRT(e.pageX, e.pageY)).then(function (invokedCommand) {
            if (invokedCommand === null) {
                // The command is null if no command was invoked.
                WinJS.log && WinJS.log("Context menu dismissed", "sample", "status");
            }
        });

        WinJS.log && WinJS.log("Context menu shown", "sample", "status");
    }

    function onOpenWith() {
        // Add command handler code here.
        WinJS.log && WinJS.log("'Open with' button clicked", "sample", "status");
    }

    function onSaveAttachment() {
        // Add command handler code here.
        WinJS.log && WinJS.log("'Save attachment' button clicked", "sample", "status");
    }
})();

File: Samples/Logging/cppwinrt/Scenario1.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario1.h"
#include "Scenario1.g.cpp"
#include "LoggingChannelScenario.h"

using namespace winrt;
using namespace winrt::SDKTemplate::implementation;

Scenario1::Scenario1()
{
    InitializeComponent();
}

void Scenario1::DoWin81Mode(Windows::Foundation::IInspectable const&, Windows::UI::Xaml::RoutedEventArgs const&)
{
    LoggingChannelScenario().LogWithWin81Constructor();
    MainPage::Current().NotifyUser(L"Complete: Windows 8 mode", NotifyType::StatusMessage);
}

void Scenario1::DoWin10Mode(IInspectable const&, Windows::UI::Xaml::RoutedEventArgs const&)
{
    LoggingChannelScenario().LogWithWin10Constructor();
    MainPage::Current().NotifyUser(L"Complete: Windows 10 mode", NotifyType::StatusMessage);
}
